diff --git a/hw/i386/microvm.c b/hw/i386/microvm.c
index bd657a9388..f442cd02d7 100644
--- a/hw/i386/microvm.c
+++ b/hw/i386/microvm.c
@@ -188,6 +188,5 @@ static void microvm_devices_init(MicrovmMachineState *mms)
 
     mms->virtio_irq_base = 5;
-    mms->virtio_num_transports = 8;
     if (x86ms->ioapic2) {
         mms->pcie_irq_base = 16;    /* 16 -> 19 */
         /* use second ioapic (24 -> 47) for virtio-mmio irq lines */
@@ -603,6 +602,24 @@ static bool microvm_machine_get_auto_kernel_cmdline(Object *obj, Error **errp)
     return mms->auto_kernel_cmdline;
 }
 
+static void microvm_machine_get_virtio_mmio_transports(Object *obj, Visitor *v,
+                                                       const char *name,
+                                                       void *opaque,
+                                                       Error **errp)
+{
+    MicrovmMachineState *mms = MICROVM_MACHINE(obj);
+    visit_type_uint32(v, name, &mms->virtio_num_transports, errp);
+}
+
+static void microvm_machine_set_virtio_mmio_transports(Object *obj, Visitor *v,
+                                                       const char *name,
+                                                       void *opaque,
+                                                       Error **errp)
+{
+    MicrovmMachineState *mms = MICROVM_MACHINE(obj);
+    visit_type_uint32(v, name, &mms->virtio_num_transports, errp);
+}
+
 static void microvm_machine_set_auto_kernel_cmdline(Object *obj, bool value,
                                                     Error **errp)
 {
@@ -625,6 +642,7 @@ static void microvm_machine_initfn(Object *obj)
     mms->ioapic2 = ON_OFF_AUTO_AUTO;
     mms->isa_serial = true;
     mms->option_roms = true;
+    mms->virtio_num_transports = 0;
     mms->auto_kernel_cmdline = true;
 }
 
@@ -713,8 +731,18 @@ static void microvm_class_init(ObjectClass *oc, const void *data)
                                    microvm_machine_get_auto_kernel_cmdline,
                                    microvm_machine_set_auto_kernel_cmdline);
     object_class_property_set_description(oc,
         MICROVM_MACHINE_AUTO_KERNEL_CMDLINE,
         "Set off to disable adding virtio-mmio devices to the kernel cmdline");
+
+    object_class_property_add(oc, MICROVM_MACHINE_VIRTIO_MMIO_TRANSPORTS,
+                              "uint32",
+                              microvm_machine_get_virtio_mmio_transports,
+                              microvm_machine_set_virtio_mmio_transports,
+                              NULL, NULL);
+    object_class_property_set_description(oc,
+        MICROVM_MACHINE_VIRTIO_MMIO_TRANSPORTS,
+        "Number of built-in virtio-mmio transports");
 
     machine_class_allow_dynamic_sysbus_dev(mc, TYPE_RAMFB_DEVICE);
+    machine_class_allow_dynamic_sysbus_dev(mc, "axi-mmio-proxy");
 
diff --git a/include/hw/i386/microvm.h b/include/hw/i386/microvm.h
index 2ad87ef551..b194649b4d 100644
--- a/include/hw/i386/microvm.h
+++ b/include/hw/i386/microvm.h
@@ -73,6 +73,7 @@
 #define MICROVM_MACHINE_ISA_SERIAL          "isa-serial"
 #define MICROVM_MACHINE_OPTION_ROMS         "x-option-roms"
 #define MICROVM_MACHINE_AUTO_KERNEL_CMDLINE "auto-kernel-cmdline"
+#define MICROVM_MACHINE_VIRTIO_MMIO_TRANSPORTS "virtio-mmio-transports"
 
 struct MicrovmMachineClass {
     X86MachineClass parent;
